### DISCUSSION TITLE
Best way to exclude a folder except certain files inside?

### DISCUSSION THREAD

**Discussion by AkazaRenn** [abraunegg/onedrive#2340] [Q&A]
Hi all! I'm trying to do what's described in the title with `sync_list`, but it is not working. Here's what it looks like:

```gitignore
/Cloud/Apps/Docker/onedrive/config
!/Cloud/Apps/Docker/onedrive
/Cloud/Apps/Docker/onedrive/sync_list
/Cloud/Apps/Docker/*
```

What I'm trying to do is to sync the configs of this OneDrive client itself but not any temp files generated. I've tried the above config but neither `config` nor `sync_list` has been uploaded, so I assume it's not a position issue. Does anyone has a suggestion on it? Thanks!

**Comment by abraunegg** [comment]
@AkazaRenn 
The use of `sync_list` excludes *everything* by default.

Try this:
```
Cloud/Apps/Docker/onedrive/config
Cloud/Apps/Docker/onedrive/sync_list
```

**Reply by AkazaRenn** [reply]
Thanks for the suggestion! Will it work with the last line that includes all other files in `Docker`?

**Comment by abraunegg** [comment]
@AkazaRenn 
What application version are you using? What is the output of `onedrive --version` ?

'sync_list' ordering matters greatly.

The issue here is - your last line is an exclusive mutual inclusion for everything, and ordering of items is specific. Please try the following:
```
!Cloud/Apps/Docker/onedrive/*
Cloud/Apps/Docker/onedrive/config
Cloud/Apps/Docker/onedrive/sync_list
Cloud/Apps/Docker/*
```

**Reply by AkazaRenn** [reply]
Unfortunately it doesn't work, I'm running `onedrive v2.4.23` inside a docker container.

**Comment by abraunegg** [comment]
@AkazaRenn 
So from what I am understanding you want to do ... you want to sync, to the cloud, the Docker Container configuration you are running ..

Docker is not going to let you access the running configuration of the container from the configuration file path .... thats a Docker security item.

You will need to backup / create a copy of those files and store them on your configured 'sync_dir' for them to be synced to OneDrive using the same Docker instance .....

**Reply by AkazaRenn** [reply]
You are right, but it is fine for the client to access these files because they are not inside the containers. These are the config files on the host machine mapped to volumes which everyone able to access that machine is able to read, so for OneDrive.

**Comment by badda71** [comment]
Any update on this? I'm trying to do the same thing ...
How exactly does the sync_list work anyways? The documentation on this is not very specific. I assume: for each file, sync_list is read from top to bottom and if the filename matches any include/exclude statement encountered, the processing stops and the file will be handled as per the statement. If nothing matches, the file is excluded. Is this correct?
If yes, the following should work to solve @AkazaRenn problem:

Cloud/Apps/Docker/onedrive/config
Cloud/Apps/Docker/onedrive/sync_list
!Cloud/Apps/Docker/onedrive/*
Cloud/Apps/Docker/*

If I want to change the default behaviour to exclude everything (so that everything is included per default), I would add a single '*' as the last line, correct?
